table/sst_file_dumper.cc

```diff
@@ -178,7 +178,10 @@ Status SstFileDumper::DumpTable(const std::string& out_filename) {
   Env* env = options_.env;
   env->NewWritableFile(out_filename, &out_file, soptions_);
   Status s = table_reader_->DumpTable(out_file.get());
-  out_file->Close();
+  if (!s.ok()) {
+    return s;
```

Comment:
Is it OK to return without closing the WritableFile? (I wish the API docs were any help 😡😡)

Reply:
Good point, I think we should try to close whether there's error or not.